ui/src/journals/containers/DetailPageContainer.tsx

```diff
@@ -0,0 +1,113 @@
+import DocumentHead from '../../common/components/DocumentHead';
+import UrlsAction from '../../literature/components/UrlsAction';
+import PublicNotesList from '../../common/components/PublicNotesList';
+import { Journal } from './SearchPageContainer';
+import { LITERATURE } from '../../common/routes';
+
+interface RootState {
+  journals: {
+    get: (values: string) => Journal;
+    hasIn: (values: string[]) => boolean;
+  };
+}
+
+export const DetailPage = ({ result }: { result: Journal }) => {
+  const [modalVisible, setModalVisible] = useState<boolean>(false);
+
+  const metadata = result.get('metadata');
+
+  const shortTitle = metadata.get('short_title');
+  const journalTitle = metadata.getIn(['journal_title', 'title']);
+  const urls = metadata.get('urls');
+  const publicNotes = metadata.get('public_notes');
+  // @ts-ignore
+  const titleVariants = metadata.get('title_variants').toArray();
+  // @ts-ignore
+  const publisher = metadata.get('publisher').toArray();
+
+  const onModalVisibilityChange = () => {
+    setModalVisible(!modalVisible);
+  };
+
+  return (
+    <>
+      <DocumentHead title={shortTitle} />
+      <Row justify="center">
+        <Col className="mv3" xs={24} md={22} lg={21} xxl={18}>
+          <ContentBox
+            className="sm-pb3"
+            leftActions={urls && <UrlsAction urls={urls} text="links" />}
+          >
+            <Row>
+              <Col>
+                <PageHeader
+                  className="site-page-header"
+                  title={shortTitle}
+                  subTitle={`(${publisher.map((p: string) => p)})`}
+                />
+              </Col>
+            </Row>
+            <Row>
+              <Col>
+                {journalTitle}
+                <Button className="btn-ghost" onClick={onModalVisibilityChange}>
+                  Show other names ({titleVariants.length})
+                </Button>
+              </Col>
+            </Row>
+            {publicNotes && (
+              <Row className="mt2">
+                <Col>
+                  <PublicNotesList publicNotes={publicNotes} />
+                </Col>
+              </Row>
+            )}
+            <Row className="mt2">
+              <Col>
+                <Link to={`${LITERATURE}?sort=mostrecent&size=25&page=1&q=refersto ${shortTitle}`}>Articles published in {shortTitle}</Link>
+              </Col>
+            </Row>
+          </ContentBox>
+        </Col>
+      </Row>
+      <Modal
+        visible={modalVisible}
+        onCancel={onModalVisibilityChange}
+        footer={null}
+      >
+        <List
+          header={<h3>Title variants</h3>}
+          size="small"
+          dataSource={titleVariants}
+          pagination={{
+            pageSize: 8,
+            size: 'small',
+            hideOnSinglePage: true
+          }}
+          renderItem={(item: string) => <List.Item>{item}</List.Item>}
+        />
+      </Modal>
+    </>
+  );
+}
```

Comment:
maybe you can extract this modal as a new component like we do for authors https://github.com/inspirehep/inspirehep/blob/ba5fe0fdd9aa4daebad985376068487edccf995a/ui/src/common/components/AuthorList.jsx?